### PR TITLE
Specs n+1

### PR DESCRIPTION
**Pull request by Juanpa8830**
# Hi reviewer.
### Make sure that the N+1 problem is solved when fetching all posts and their comments for a user:

- Before applying changes:

image

- After applying changes:
image

### Use Capybara to write integration tests for each view in your project.

Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

#### User index page:

- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.

#### user show page:

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click to see all posts, it redirects me to the user's post's index page.

#### User post index page:

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

#### Post show page:

- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  it 'I can see a button that lets me view all of a user\'s posts.' do
+    expect(page).to have_content('See all posts')
+  end
+
+  it "When I click to see all posts, it redirects me to the user's post's index page." do
+    click_link 'See all posts'
+    expect(page).to have_current_path(user_posts_path(@user1))
+  end
+end
```
Comment (julie-ify) [inline spec/views/user_show_spec.rb:50]: - [x] Kindly add test for this assertion:

  - Tests that redirects to the post's show page when you click on a user's post.

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  it 'I can see a section for pagination if there are more posts than fit on the view.' do
+    expect(page).to have_content('Pagination')
+  end
+
+  it 'When I click on a post, it redirects me to that posts show page.' do
+    click_link "Post ##{@post1.id}"
+    expect(page).to have_current_path(user_post_path(@user1, @post1))
+  end
+end
```
Comment (julie-ify) [inline spec/views/post_index_spec.rb]: - [x] Kindly add test for this scenario:

  - Tests that asserts you can see how many likes a post has

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Dadadon and @Juanpa8830,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_show_spec.rb (hunk 1 above)
Comment (Juanpa8830) [reply spec/views/user_show_spec.rb:50]: @julie-ify Thank you for your comments. I'd like to ask you if this test is not a part of the post_index view. In this case I created this test in the post_index_spec.rb file like 'When I click on a post, it redirects me to that posts show page.'

**Review by julie-ify — APPROVED** [review]
Hi @Juanpa8830 and @Dadadon,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**